test/backup_only.py

```diff
@@ -0,0 +1,401 @@
+    with open(environment.vttop + '/config/init_db.sql') as fd:
+      init_db = fd.read()
+    with open(new_init_db, 'w') as fd:
+      fd.write(init_db)
+      fd.write('''
+# Set real passwords for all users.
+ALTER USER 'root'@'localhost' IDENTIFIED BY 'RootPass';
+ALTER USER 'vt_dba'@'localhost' IDENTIFIED BY 'VtDbaPass';
+ALTER USER 'vt_app'@'localhost' IDENTIFIED BY 'VtAppPass';
+ALTER USER 'vt_allprivs'@'localhost' IDENTIFIED BY 'VtAllPrivsPass';
+ALTER USER 'vt_repl'@'%' IDENTIFIED BY 'VtReplPass';
+ALTER USER 'vt_filtered'@'localhost' IDENTIFIED BY 'VtFilteredPass';
+FLUSH PRIVILEGES;
+''')
+    credentials = {
+        'vt_dba': ['VtDbaPass'],
+        'vt_app': ['VtAppPass'],
+        'vt_allprivs': ['VtAllprivsPass'],
+        'vt_repl': ['VtReplPass'],
+        'vt_filtered': ['VtFilteredPass'],
+    }
+    db_credentials_file = environment.tmproot+'/db_credentials.json'
+    with open(db_credentials_file, 'w') as fd:
+      fd.write(json.dumps(credentials))
+    logging.debug("initilizing mysql %s",str(datetime.datetime.now()))
+    # start mysql instance external to the test
+    setup_procs = [
+        tablet_master.init_mysql(init_db=new_init_db,
+                                 extra_args=['-db-credentials-file',
+                                             db_credentials_file]),
+        tablet_replica1.init_mysql(init_db=new_init_db,
+                                   extra_args=['-db-credentials-file',
+                                               db_credentials_file]),
+        tablet_replica2.init_mysql(init_db=new_init_db,
+                                   extra_args=['-db-credentials-file',
+                                               db_credentials_file])
+    ]
+    if use_mysqlctld:
+      tablet_master.wait_for_mysqlctl_socket()
+      tablet_replica1.wait_for_mysqlctl_socket()
+      tablet_replica2.wait_for_mysqlctl_socket()
+    else:
+      utils.wait_procs(setup_procs)
+    logging.debug("done initilizing mysql %s",str(datetime.datetime.now()))      
+  except:
+    tearDownModule()
+    raise
+
+
+def tearDownModule():
+  utils.required_teardown()
+  if utils.options.skip_teardown:
+    return
+
+  teardown_procs = [
+      tablet_master.teardown_mysql(extra_args=['-db-credentials-file',
+                                               db_credentials_file]),
+      tablet_replica1.teardown_mysql(extra_args=['-db-credentials-file',
+                                                 db_credentials_file]),
+      tablet_replica2.teardown_mysql(extra_args=['-db-credentials-file',
+                                                 db_credentials_file])
+  ]
+  utils.wait_procs(teardown_procs, raise_on_error=False)
+
+  environment.topo_server().teardown()
+  utils.kill_sub_processes()
+  utils.remove_tmp_files()
+
+  tablet_master.remove_tree()
+  tablet_replica1.remove_tree()
+  tablet_replica2.remove_tree()
+  backup_tablet.remove_tree()
+
+class TestBackup(unittest.TestCase):
```

Comment:
The class name here is the same as in backup.py. Is that legal?

Reply:
Yeah, they just show up as a similar name in the test logs. You have to specify each test case in the config.json file. These names are for logging only